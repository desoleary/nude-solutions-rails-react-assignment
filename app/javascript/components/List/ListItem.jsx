import { List, Typography } from 'antd';
import { arrayOf, node, number, oneOfType, string } from 'prop-types';
import React from 'react';

const { Item } = List;
const { Text } = Typography;

Item.displayName = 'ItemAntd';

const ListItem = ({ text, extra, actions, id }) => {
  const key = `list-item-${id}`;

  return (
    <Item extra={extra} actions={actions} key={key}>
      <Text>{text}</Text>
    </Item>
  );
};

ListItem.defaultProps = {
  actions: []
};

ListItem.propTypes = {
  id: oneOfType([number, string]).isRequired,
  text: string,
  extra: node,
  actions: arrayOf(node)
};

export default ListItem;
