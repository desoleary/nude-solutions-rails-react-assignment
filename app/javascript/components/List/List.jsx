import { List as ListAntd } from 'antd';
import { arrayOf, node, number, oneOfType, shape, string } from 'prop-types';
import React from 'react';
import ListItem from './ListItem';

ListAntd.displayName = 'ListAntd';

const List = (props) => {
  const { dataSource } = props;

  return (
    <ListAntd
      dataSource={dataSource}
      size='small'
      split={false}
      renderItem={ListItem}
    />
  );
};

List.propTypes = {
  dataSource: arrayOf(
    shape({
      id: oneOfType([number, string]).isRequired,
      text: string,
      extra: node,
      actions: arrayOf(node)
    })
  )
};

export default List;
