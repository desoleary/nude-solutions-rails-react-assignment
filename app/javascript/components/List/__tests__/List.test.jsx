import { shallow } from 'enzyme';
import React from 'react';
import { List as ListAntd } from 'antd';
import List from '../List';

describe('with shallow rendering', () => {
  let wrapper;

  const listItemProps = {
    id: 'some-identifier',
    text: 'some-text',
    extra: <div>some-extra-content</div>,
    actions: [<div>some-action-to-perform</div>]
  };

  const props = {
    dataSource: [listItemProps]
  };

  beforeEach(() => {
    wrapper = shallow(<List {...props} />);
  });

  it('compares snapshot', () => {
    expect(wrapper.length).to.eql(1);
    expect(wrapper).toMatchSnapshot();
  });

  it('passes props correctly', () => {
    const antdListProps = wrapper.find(ListAntd).props();

    expect(antdListProps.size).to.eql('small');
    expect(antdListProps.split).to.eql(false);
    expect(antdListProps.dataSource).to.eql([listItemProps]);
  });
});
