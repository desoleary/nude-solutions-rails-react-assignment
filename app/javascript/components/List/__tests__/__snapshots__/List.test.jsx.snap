// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`with shallow rendering compares snapshot 1`] = `
<ListAntd
  dataSource={
    Array [
      Object {
        "actions": Array [
          <div>
            some-action-to-perform
          </div>,
        ],
        "extra": <div>
          some-extra-content
        </div>,
        "id": "some-identifier",
        "text": "some-text",
      },
    ]
  }
  renderItem={[Function]}
  size="small"
  split={false}
/>
`;
